### ISSUE TITLE
Issues calling the Cardano API on Windows

### ISSUE THREAD

**Issue by jmevel** [input-output-hk/daedalus#1942]
Hello guys,

I'm trying to send a `curl` request to my local Cardano node's API but I'm facing an issue regarding the rights to access it.

I opened a Windows Command Prompt as Administrator and moved to this directory `C:\Users\jerome\AppData\Roaming\Daedalus Flight\tls\client` then I executed the following command:

```bash
curl -X GET https://localhost:8090/api/v1/node-info -H "Accept: application/json;charset=utf-8" --cacert ca.crt --cert client.pem
```

But I get a nice `curl: (7) Failed to connect to localhost port 8090: Connection refused` error.

So I double-clicked on both `ca.crt` and `client.crt` to install them in the Windows Certificate Store under `Trusted Root Certification Authorities` (I can then view them successfully installed using the Microsoft Management Console).
But even after that it still doesn't work (I even tried to install the files located in the `server` directory as well)

I've seen [here](https://github.com/input-output-hk/daedalus/issues/1240) that Daedalus is now using a dynamic port to expose the Cardano server so I downloaded the logs, opened the `Daedalus.json` file and looked for the string `ReplyPort`. There was multiple values found in the document so I used the latest one:

```bash
curl -X GET https://localhost:64063/api/v1/node-info -H "Accept: application/json;charset=utf-8" --cacert ./ca.crt --cert ./client.pem --ssl-no-revoke
```

This results with an error
> curl: (35) schannel: next InitializeSecurityContext failed: SEC_E_CERT_UNKNOWN (0x80090327) - An unknown error occurred while processing the certificate.

Note: I added `--ssl-no-revoke` after a while otherwise I'd get the error `The revocation function was unable to check revocation for the certificate`

As a last attempt, as I saw in the other GitHub issue I linked above, I should be able to use the `-k` parameter (`--insecure`) to ignore certificate validation but I get the same error

```bash
curl -k https://localhost:64063/api/v1/node-info -H "Accept: application/json;charset=utf-8"
```

Could you please let me know how to properly call the Cardano server's API? Should I use port `8090` or the dynamic one?
How to get past this damn certificate issue?
Also last question: why when I open Cardano Flight and toggle the Developer Console I can't see any HTTP request made to the local Cardano server. How come? Am I not supposed to see requests to the different localhost endpoints? I even tried to catch the requests using Fiddler but I see nothing...

Thanks for your answers, I'm trying to recover my wallet. I'm starting with a very basic GET request but my goal is to call `api/internal/import-wallet` with my `secret.key` file and my spending password to restore my wallet. (for some reason I saved the wrong recovery phrase)

**Comment by nikolaglumac** [comment]
Hi @jmevel!
Daedalus Flight uses v2 Api and this is why your v1 Api calls are failing.
Here are the docs where you can find more info about the available endpoints: https://input-output-hk.github.io/cardano-wallet/api/edge/
Please look under "Byron" section. Thanks!

**Comment by jmevel** [comment]
Ok thanks for your answer.

So finally I used Daedalus 0.15.1 because on the API v2 I can't see any endpoint to *Import a wallet from Disk* [like in the API v1.7.0](https://docs.cardano.org/technical/wallet/api/v1/?v=1.7.0#tag/Internal%2Fpaths%2F~1api~1internal~1import-wallet%2Fpost).
Is it a deprecated feature? I noticed in Daedalus, the big button to *Import a wallet from a file* is disabled...

At the end my attempt was a fail. If I call the API with the original spending password I set few years ago

```bash
curl -X POST https://localhost:50691/api/internal/import-wallet -H "Accept: application/json;charset=utf-8" -H "Content-Type: application/json" --cacert ca.crt --cert client.pem --ssl-no-revoke -d "{\"filePath\": \"C:\\TestImportWallet\\secret.key\",\"spendingPassword\": \"MyOriginalPassword\"}"
```

I get an error saying `The system cannot find the path specified.`. I tried to run `cardano-launcher.exe` as Administrator or giving a relative path and copy this `secret.key` file in `C:\Program Files\Daedalus` but it doesn't work.

If I make a call without providing the spending password I get another error 
```json
{"status":"error","diagnostic":{"msg":"ImportWalletNoWalletFoundInBackup C:\\TestImportWallet\\secret.key"},"message":"UnknownError"}
```

I guess my Cardano are lost forever then. 
Too bad 😥
